### ISSUE TITLE
lxc-attach can not exit when active detached sub-processes exists

### ISSUE THREAD

**Issue by asainkujovic** [lxc/lxc#4507]
* Distribution: any, as far as I know this behaviour did not changed for years and years
 
# issue description:

If you run lxc-attach and inside it if you run some detached process like for example "tail -f somelog.txt &", and than continue to work inside lxc-attach shell, and then you try to exit ... you are blocked, you can not use that terminal anymore. Shell has subprocess and on exit shell enters some zombie 'defunct' state, child subprocess goes to parent 'init (1)', but parent of parent, our lxc-attach does nothing about it, just wait and wait.

# Steps to reproduce

1. run `lxc-attach name`
2. run `sleep 25 &` inside it
3. run `exit` and you will be blocked for25 seconds

# Information to attach

After inspecting this problem when I tried to make minor code changes to test it, i found out that good thing is that lxc-attach in 'exit' moment recive 'SIGCHLD' well. So if we add somewhere at 1734 line of file attach.c line like: `signal(SIGCHLD, newhandler)`,
problem would be solved if we just also define this `void newhandler(int)` as something like: `{ if waitpid((-1,NULL,WNOHANG)>0) exit(0); }`. In this code, waitpid(-1) used only to get childpid number, and if it is 0 - avoid killing lxc-attach process in the beginning (0 initial fork) ... just ignore pid0, and exit for other (other is the only child process that lxc-attach has, shell itself. When shell sends signal 'child is killed', then lxc-attach may also exit.

Not sure that this is ideal solution but it shows me that problem could be solved with ease for sure. So hope that someone from creators of lxc can check this, and i am also wondering why this problem was not listed before, because it is very annoying in manual usage but also block us often to use lxc-attach integrated inside something else.

**Comment by hallyn** [comment]
Hm.  But is that not the behavior we want?

@stgraber @brauner I'm curious of your opinion here.  I think this behavior is actually preferable to not waiting for all processes to exit.

**Comment by asainkujovic** [comment]
@hallyn, yes, it should be behaviour that any user want and expect, to exit when typing 'exit' without hanging in blocked program forever (or for 25 seconds in above example). Current version without the patch from [connected PR](https://github.com/lxc/lxc/pull/4509) and all versions for last 5 years that I checked will be blocked for sure.

**Comment by asainkujovic** [comment]
closing issue because PR [#4509](https://github.com/lxc/lxc/pull/4509) that should solve this problem is now merged well

**Comment by hallyn** [comment]
As @mihalicyn mentioned, let's introduce a new attach argument for this behavior.
